### PR TITLE
feat(trajectory_follower): pub steer converged marker

### PR DESCRIPTION
**Pull request by kosuke55**
Signed-off-by: kosuke55 <kosuke.tnp@gmail.com>

## Description

pub steeer conveged info as debug marker array.

image

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp [hunk 1]
```diff
@@ -187,7 +191,7 @@ class TRAJECTORY_FOLLOWER_PUBLIC MpcLateralController : public LateralController
   /**
    * @brief check ego car is in stopped state
    */
-  bool isStoppedState() const;
+  bool isStoppedState();
```
Comment (takayuki5168) [inline control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp:194]: No need to remove const?

File: control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp [hunk 2]
```diff
@@ -133,6 +134,9 @@ class TRAJECTORY_FOLLOWER_PUBLIC MpcLateralController : public LateralController
   double m_ego_nearest_dist_threshold;
   double m_ego_nearest_yaw_threshold;
 
+  // debug marker
+  visualization_msgs::msg::MarkerArray m_debug_marker_array;
```
Comment (takayuki5168) [inline control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp:138]: I don't think adding a member variable which can be a local variable is a good way.
What do you think?

Comment (kosuke55) [reply control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp:138]: I made it as a member for considering adding other markers later. 
But in this PR, only the maker for steer converges, so it's also ok to use a local variable.

File: control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp (hunk 1 above)
Comment (kosuke55) [reply control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp:194]: `isSteerConverged` is used in this funcion.
and `isSteerConverged` adds a marker to a member variable. 
so removing const is needed if making `m_debug_marker_array` a member.

File: control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp (hunk 2 above)
Comment (takayuki5168) [reply control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp:138]: If this variable really has to be global later, you don't have to make it local.

Comment (kosuke55) [reply control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp:138]: No real plans, just a possibility.

**Review by brkay54 — CHANGES_REQUESTED** [review]
Thank you for your work. Code looks good to me. 

Only problem I see that because `isSteerConverged()` function called twice in `run()` function, you are pushing same marker twice. It causes error in rviz:

Screenshot from 2022-12-04 22-11-56

Also, can you remap the debug topic in `control.launch.py` (also you can create another PR for this.)

File: control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp (hunk 2 above)
Comment (kosuke55) [reply control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp]: Moved these markers to interface nodes.

Comment (kosuke55) [reply control/trajectory_follower/include/trajectory_follower/mpc_lateral_controller.hpp]: so removed member markers from lateral controller.

**Comment by kosuke55** [comment]
@brkay54 
sorry to be late and thanks for your feedback.
I discussed with @takayuki5168 and we decided to move these makers to interface `controller_node`.
It only publishes the marker once, so I think the problem was solved.

File: control/trajectory_follower/src/mpc_lateral_controller.cpp [hunk 3]
```diff
@@ -234,6 +234,7 @@ bool MpcLateralController::isSteerConverged(
   // wait for a while to propagate the trajectory shape to the output command when the trajectory
   // shape is changed.
   if (!m_has_received_first_trajectory || isTrajectoryShapeChanged()) {
+    RCLCPP_DEBUG(node_->get_logger(), "trajectory shaped is changed");
```
Comment (kosuke55) [inline control/trajectory_follower/src/mpc_lateral_controller.cpp:237]: `controller_node' can not get this information, so just print debug message.

**Review by takayuki5168 — APPROVED** [review]
LGTM

**Comment by kosuke55** [comment]
@brkay54 
Could you please withdraw the change request if no problem?:pray:

**Pull request merged**